plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -866,58 +848,6 @@ public void abortReadConnection(Connection connection, ResultSet resultSet)
         }
     }
 
-    @Override
-    public Connection getConnection(ConnectorSession session, JdbcOutputTableHandle handle)
-            throws SQLException
-    {
-        return configureConnectionTransactionIsolation(super.getConnection(session, handle));
-    }
-
-    @Override
-    public Connection getConnection(ConnectorSession session, JdbcSplit split)
-            throws SQLException
-    {
-        return configureConnectionTransactionIsolation(super.getConnection(session, split));
-    }
-
-    private Connection configureConnectionTransactionIsolation(Connection connection)
-            throws SQLException
-    {
-        if (snapshotIsolationDisabled) {
-            return connection;
-        }
-        try {
-            if (hasSnapshotIsolationEnabled(connection)) {
-                // SQL Server's READ COMMITTED + SNAPSHOT ISOLATION is equivalent to ordinary READ COMMITTED in e.g. Oracle, PostgreSQL.
-                connection.setTransactionIsolation(TRANSACTION_SNAPSHOT);
-            }
-        }
-        catch (SQLException e) {
-            connection.close();
-            throw e;
-        }
-
-        return connection;
-    }
-
-    private boolean hasSnapshotIsolationEnabled(Connection connection)
-            throws SQLException
-    {
-        try {
-            return snapshotIsolationEnabled.get(SnapshotIsolationEnabledCacheKey.INSTANCE, () -> {
-                Handle handle = Jdbi.open(connection);
-                return handle.createQuery("SELECT snapshot_isolation_state FROM sys.databases WHERE name = :name")
```

Comment:
Why remove this? I think this was so that we can adapt to remote system's configuration without redundant configuration.

Reply:
The logic is moved to `SqlServerConnectionFactory`, not deletion. 
https://github.com/trinodb/trino/pull/11662/files#diff-73b2f26b2b5719c3dc9f5dcc10865f0537b0f886558b90784ea2fef3a9f6fd3bR94